src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -1216,7 +1365,7 @@ internal bool TryGetGuidCore(out Guid value)
                 }
 
                 Debug.Assert(sequenceLength <= JsonConstants.MaximumEscapedGuidLength);
-                Span<byte> stackSpan = stackalloc byte[_stringHasEscaping ? JsonConstants.MaximumEscapedGuidLength : JsonConstants.MaximumFormatGuidLength];
+                Span<byte> stackSpan = stackalloc byte[ValueIsEscaped ? JsonConstants.MaximumEscapedGuidLength : JsonConstants.MaximumFormatGuidLength];
```

Comment:
```suggestion
                Span<byte> stackSpan = ValueIsEscaped ? stackalloc byte[JsonConstants.MaximumEscapedGuidLength] : stackalloc byte[JsonConstants.MaximumFormatGuidLength];
```

Reply:
We follow this pattern fairly pervasively, so it might make sense to make this change everywhere in a separate PR?

Reply:
Sure.  It could also not even branch and just pick the larger value to always use.

Reply:
Addressed in https://github.com/dotnet/runtime/pull/69580/commits/9233304dd19275ef2ce25898378617ecac0b6d39